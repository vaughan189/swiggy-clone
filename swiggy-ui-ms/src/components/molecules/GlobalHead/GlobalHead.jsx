/**
 *
 * GlobalHead
 *
 */
import React from 'react';

import Favicon from '../Favicon';
import PreConnects from '../Preconnects';

const GlobalHead = () => {
  return (
    <>
      <title>Swiggy Clone</title>
      <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" />
      <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
      <PreConnects />
      <Favicon />
         <link
            rel="stylesheet"
            href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css"
            integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2"
            crossOrigin="anonymous"
        />
        <script
            src="https://kit.fontawesome.com/c34fec7b4b.js"
            crossOrigin="anonymous"
        ></script>

        <link
            rel="stylesheet"
            href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css"
        />
         <script
            src="https://code.jquery.com/jquery-3.5.1.slim.min.js"
            integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj"
            crossOrigin="anonymous"
        ></script>
        <script
            src="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/js/bootstrap.bundle.min.js"
            integrity="sha384-ho+j7jyWK8fNQe+A12Hb8AhRq26LrZ/JpcUGGOn+Y7RsweNrtN/tE3MoK7ZeZDyx"
            crossOrigin="anonymous"
        ></script>
    </>
  );
};

export default GlobalHead;
