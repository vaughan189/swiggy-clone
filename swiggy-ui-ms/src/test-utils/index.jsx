import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
        {children}
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
