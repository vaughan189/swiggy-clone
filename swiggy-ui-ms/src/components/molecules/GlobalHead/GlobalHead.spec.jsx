import React from 'react';
import { render } from '../../../test-utils';

import GlobalHead from '.';

describe('GlobalHead', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<GlobalHead />);
    expect(asFragment()).toMatchSnapshot();
  });
});
