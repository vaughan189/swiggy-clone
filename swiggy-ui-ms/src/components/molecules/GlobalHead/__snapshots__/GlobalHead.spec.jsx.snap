// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalHead should render correctly 1`] = `
<DocumentFragment>
  <title>
    Swiggy Clone
  </title>
  <link
    href="https://fonts.googleapis.com/css?family=Roboto:300,400,500"
    rel="stylesheet"
  />
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
    sizes="96x96"
    type="image/png"
  />
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/ag-grid/19.1.4/styles/ag-grid.css"
    rel="stylesheet"
  />
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/ag-grid/19.1.4/styles/ag-theme-balham.css"
    rel="stylesheet"
  />
  <link
    href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
    rel="stylesheet"
  />
</DocumentFragment>
`;
